docs/src/main/sphinx/connector/hive.md

```diff
@@ -628,14 +628,41 @@ Hive directly, most operations can be performed using Trino.
 
 #### Schema evolution
 
-Hive allows the partitions in a table to have a different schema than the
-table. This occurs when the column types of a table are changed after
-partitions already exist (that use the original column types). The Hive
-connector supports this by allowing the same conversions as Hive:
-
-- `VARCHAR` to and from `TINYINT`, `SMALLINT`, `INTEGER` and `BIGINT`
-- `REAL` to `DOUBLE`
-- Widening conversions for integers, such as `TINYINT` to `SMALLINT`
+Hive table partitions can differ from the current table schema. This occurs when
+the data types of columns of a table are changed from the data types of columns
+of preexisting partitions. The Hive connector supports this schema evolution by
+allowing the same conversions as Hive. The following table lists possible data
+type conversions.
+
+:::{list-table} Hive schema evolution type conversion
+:widths: 25, 75
+:header-rows: 1
+
+* - Data type
+  - Converted to
+* - `VARCHAR`
+  - `TINYINT`, `SMALLINT`, `INTEGER`, `BIGINT`, `TIMESTAMP`, `DATE`, as well as
+    narrowing conversions for `VARCHAR`
+* - `TINYINT`
+  - `VARCHAR`, as well as widening conversions, such as `TINYINT` to `SMALLINT`
+* - `SMALLINT`
+  - `VARCHAR`, as well as widening conversions
+* - `INTEGER`
+  - `VARCHAR`, as well as widening conversions
+* - `BIGINT`
+  - `VARCHAR`
+* - `REAL`
+  - `DOUBLE`, `DECIMAL`
+* - `CHAR`
+  - narrowing conversions for `CHAR`
```

Comment:
Move `CHAR` to be below `VARCHAR` so we have:
* character types
* numeric types
* temporal types